### ISSUE TITLE
glm-ocr sft 的bug

### ISSUE THREAD

**Issue by Bourne-M** [modelscope/ms-swift#8124]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

glmocr用了GLM4TemplateMeta，这里会给每条样本加上【<|user|>】，并且loss权重为1.0
extra_context_list = template_meta.suffix
extra_context_type = ContextType.SUFFIX

### How to Reproduce / 如何复现

CUDA_VISIBLE_DEVICES=0 \
swift sft \
    --model ZhipuAI/GLM-OCR \
    --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite
#20000
' \
    --load_from_cache_file true \
    --split_dataset_ratio 0.01 \
    --tuner_type lora \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 1e-4 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit true \
    --freeze_aligner true \
    --gradient_accumulation_steps 16 \
    --eval_steps 50 \
    --save_steps 50 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 4096 \
    --output_dir output \
    --warmup_ratio 0.05 \
    --dataset_num_proc 4 \
    --dataloader_num_workers 4

**Comment by slin000111** [comment]
> ### Checklist / 检查清单
> * [x]  I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。
> 
> ### Bug Description / Bug 描述
> glmocr用了GLM4TemplateMeta，这里会给每条样本加上【<|user|>】，并且loss权重为1.0 extra_context_list = template_meta.suffix extra_context_type = ContextType.SUFFIX
> 
> ### How to Reproduce / 如何复现
> CUDA_VISIBLE_DEVICES=0 swift sft --model ZhipuAI/GLM-OCR --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite #20000 ' --load_from_cache_file true --split_dataset_ratio 0.01 --tuner_type lora --torch_dtype bfloat16 --num_train_epochs 1 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --learning_rate 1e-4 --lora_rank 8 --lora_alpha 32 --target_modules all-linear --freeze_vit true --freeze_aligner true --gradient_accumulation_steps 16 --eval_steps 50 --save_steps 50 --save_total_limit 2 --logging_steps 5 --max_length 4096 --output_dir output --warmup_ratio 0.05 --dataset_num_proc 4 --dataloader_num_workers 4
> 
> ### Additional Information / 补充信息
> _No response_

贴一下训练的数据样例。

**Comment by Bourne-M** [comment]
> > ### Checklist / 检查清单
> > 
> > 
> > * [x]   I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。
> > ### Bug Description / Bug 描述
> > glmocr用了GLM4TemplateMeta，这里会给每条样本加上【<|user|>】，并且loss权重为1.0 extra_context_list = template_meta.suffix extra_context_type = ContextType.SUFFIX
> > ### How to Reproduce / 如何复现
> > CUDA_VISIBLE_DEVICES=0 swift sft --model ZhipuAI/GLM-OCR --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite #20000 ' --load_from_cache_file true --split_dataset_ratio 0.01 --tuner_type lora --torch_dtype bfloat16 --num_train_epochs 1 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --learning_rate 1e-4 --lora_rank 8 --lora_alpha 32 --target_modules all-linear --freeze_vit true --freeze_aligner true --gradient_accumulation_steps 16 --eval_steps 50 --save_steps 50 --save_total_limit 2 --logging_steps 5 --max_length 4096 --output_dir output --warmup_ratio 0.05 --dataset_num_proc 4 --dataloader_num_workers 4
> > ### Additional Information / 补充信息
> > _No response_
> 
> 贴一下训练的数据样例。

当msg=[{'role': 'user', 'content': '<image>Using LaTeX to perform OCR on the image.', 'loss': None}, {'role': 'assistant', 'content': '0 = \\frac { d ^ { 2 } y } { d x ^ { 2 } }', 'loss': None}] 时：

res_context_list=
['[gMASK]<sop><|user|>\n', [-100], 'Using LaTeX to perform OCR on the image.<|assistant|>\n', '0 = \\frac { d ^ { 2 } y } { d x ^ { 2 } }<|user|>']

loss_scale_list=
[0.0, 0.0, 0.0, 1.0]

**Comment by slin000111** [comment]
> ### Checklist / 检查清单
> * [x]  I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。
> 
> ### Bug Description / Bug 描述
> glmocr用了GLM4TemplateMeta，这里会给每条样本加上【<|user|>】，并且loss权重为1.0 extra_context_list = template_meta.suffix extra_context_type = ContextType.SUFFIX
> 
> ### How to Reproduce / 如何复现
> CUDA_VISIBLE_DEVICES=0 swift sft --model ZhipuAI/GLM-OCR --dataset 'AI-ModelScope/LaTeX_OCR:human_handwrite #20000 ' --load_from_cache_file true --split_dataset_ratio 0.01 --tuner_type lora --torch_dtype bfloat16 --num_train_epochs 1 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --learning_rate 1e-4 --lora_rank 8 --lora_alpha 32 --target_modules all-linear --freeze_vit true --freeze_aligner true --gradient_accumulation_steps 16 --eval_steps 50 --save_steps 50 --save_total_limit 2 --logging_steps 5 --max_length 4096 --output_dir output --warmup_ratio 0.05 --dataset_num_proc 4 --dataloader_num_workers 4
> 
> ### Additional Information / 补充信息
> _No response_

transformers推理看了一下，生成的最后一个token确实是`<|user|>`，swift的template和transformers对齐的。
